sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java

```diff
@@ -24,20 +25,21 @@
  * for both of these will be normalized to be all lower case, and underscores will be replaced with
  * periods.
  */
-public final class ConfigProperties {
+final class DefaultConfigProperties implements ConfigProperties {
```

Comment:
As in all exceptional cases this class throws standardised ConfigurationException - perhaps it would make sense to set this as a part of contract and move to ConfigProperties interface? Then of course all other implementation would be required to throw this exception in case of non-recoverable issues.

Reply:
Yeah that was the other option, and really then we may as well move the class into the SPI instead of separating an interface for `ConfigProperties`. @kenfinnigan would you want to use this default configproperties or implement it yourself? If latter, I think not pushing the exception contract to SPI makes sense - the exception is used by our autoconfiguration but doesn't have to be used by other mechanisms.

Reply:
@anuraaga Right now I think I'd be ok with using `ConfigProperties` directly, but I haven't directly played with it to know whether there would be issues I'm not thinking of

Reply:
I think I'm leaning towards what we have now - an interface is an interface, even SPI. I don't think we should have any implementation in it.